iree/compiler/InputConversion/MHLO/MHLOToLinalgOnTensors.cpp

```diff
@@ -268,6 +268,101 @@ struct ConstOpConversion : public OpConversionPattern<mhlo::ConstOp> {
+    auto targetRank = targetTy.getRank();
+    Value output;
+    Value initTensor = rewriter.create<linalg::InitTensorOp>(
+        loc, ArrayRef<Value>({}), targetTy.getShape(),
+        targetTy.getElementType());
+    auto resultTy = this->typeConverter->convertType(op.getResult().getType())
+                        .cast<ShapedType>();
+    // Creates index map using target matrix's rank.
+    SmallVector<AffineMap, 3> indexingMaps;
+    indexingMaps.push_back(AffineMap::get(targetRank, /*symbolCount=*/0,
+                                          SmallVector<AffineExpr>({}),
+                                          rewriter.getContext()));
+    indexingMaps.push_back(AffineMap::get(targetRank, /*symbolCount=*/0,
+                                          SmallVector<AffineExpr>({}),
+                                          rewriter.getContext()));
+    SmallVector<AffineExpr> outputExprs;
+    for (int i = 0; i < targetRank; i++) {
+      outputExprs.push_back(rewriter.getAffineDimExpr(i));
+    }
+    indexingMaps.push_back(AffineMap::get(targetRank, /*symbolCount=*/0,
+                                          outputExprs, rewriter.getContext()));
+    // Generic region with LCG Algorithm that make use of element index from:
+    // https://reviews.llvm.org/D101364
+    auto linalgOp = rewriter.create<linalg::GenericOp>(
+        op.getLoc(), /*resultTensors=*/resultTy,
+        /*inputs=*/ValueRange{args[0], args[1]},
+        /*outputs=*/initTensor, indexingMaps,
+        GetParallelAndReductionIterators(/*nLoops=*/targetRank,
+                                         /*nReduction=*/0),
+        [&](OpBuilder &b, Location loc, ValueRange args) {
+          llvm::SmallVector<Value> updateVec = {
+              b.create<ConstantOp>(loc, b.getI32IntegerAttr(0))};
+          Value multiplier =
+              b.create<ConstantOp>(loc, b.getI32IntegerAttr(1103515245));
+          Value incrementStep =
+              b.create<ConstantOp>(loc, b.getI32IntegerAttr(12345));
+          // For output matrix with rank N:
+          // temp1 = (cast(I32, index(D.0)) + seed) * mult + incr
+          // ...
+          // tempN = (cast(I32, index(D.(N))) + tempN_1) * mult + incr
+          for (int rank_it = 0; rank_it < targetRank; rank_it++) {
+            Value update = updateVec.back();
+            Value ind = b.create<linalg::IndexOp>(loc, rank_it);
+            Value castInd = b.create<IndexCastOp>(loc, int32Type, ind);
+            Value addRes = b.create<AddIOp>(loc, castInd, update);
+            Value multRes = b.create<MulIOp>(loc, addRes, multiplier);
+            Value incRes = b.create<AddIOp>(loc, multRes, incrementStep);
+            updateVec.push_back(incRes);
+          }
+          // Scaling = (max - min) * const(F64, 2.3283064e-10)
+          Value epsilon = b.create<ConstantOp>(
+              loc, b.getFloatAttr(args[0].getType(), 2.3283063999999999E-10));
```

Comment:
How is `2.3283064e-10` derived?

Reply:
It is 1/(4294967295) where MAX_INT = 4294967295.

Reply:
coming from random(min,max) = rng_algo()/(INT_MAX/(max-min))